packages/resolver/lib/resolver.ts

```diff
@@ -3,17 +3,26 @@ const expect = require("@truffle/expect");
 const provision = require("@truffle/provisioner");
 
 import { ResolverSource } from "./source";
-import { constructSources } from "./sources";
+import { EthPMv1, NPM, GlobalNPM, FS, Truffle } from "./sources";
 
 export class Resolver {
   options: any;
   sources: ResolverSource[];
 
-  constructor(options: any) {
+  constructor(options: any, includeTruffleSource: boolean) {
     expect.options(options, ["working_directory", "contracts_build_directory"]);
 
     this.options = options;
-    this.sources = constructSources(options);
+    this.sources = [
+      new EthPMv1(options.working_directory),
+      new NPM(options.working_directory),
+      new GlobalNPM(),
+      new FS(options.working_directory, options.contracts_build_directory)
+    ];
+
+    if (includeTruffleSource) {
+      this.sources.push(new Truffle(options));
```

Comment:
Does this need to come first? Shouldn't it have higher precedent than the others?

Reply:
I'm not sure which is better. It probably should come first I guess.